bot.js

```diff
@@ -58,110 +80,197 @@ const parseCommand = (text) => {
-                }
-            });
-            // Send a message showing all the levels gained
-            switch (Object.keys(diff).length) {
-                case 0:
-                    break;
-                case 1:
-                    const skill = Object.keys(diff)[0];
+        });
+        // Send a message showing all the levels gained
+        switch (Object.keys(diff).length) {
+            case 0:
+                break;
+            case 1:
+                const skill = Object.keys(diff)[0];
+                const levelsGained = diff[skill];
+                sendUpdateMessage(trackingChannel,
+                    `**${player}** has gained `
+                        + (levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`)
+                        + ` in **${skill}** and is now level **${newLevels[skill]}**`,
+                    skill);
+                break;
+            default:
+                const text = Object.keys(diff).toSortedSkills().map((skill) => {
                     const levelsGained = diff[skill];
-                    sendUpdateMessage(trackingChannel,
-                        `**${player}** has gained `
-                            + (levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`)
-                            + ` in **${skill}** and is now level **${newLevels[skill]}**`,
-                        skill);
-                    break;
-                default:
-                    const text = Object.keys(diff).toSortedSkills().map((skill) => {
-                        const levelsGained = diff[skill];
-                        return `${levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`} in **${skill}** and is now level **${newLevels[skill]}**`;
-                    }).join('\n');
-                    sendUpdateMessage(trackingChannel, `**${player}** has gained...\n${text}`, 'overall');
-                    break;
+                    return `${levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`} in **${skill}** and is now level **${newLevels[skill]}**`;
+                }).join('\n');
+                sendUpdateMessage(trackingChannel, `**${player}** has gained...\n${text}`, 'overall');
+                break;
+        }
+    }
+    // If not spoofing the diff, update player's levels
+    if (!spoofedDiff) {
+        levels[player] = newLevels;
+        lastUpdate[player] = new Date();
+    }
+};
+
+const updateKillCounts = (player, killCounts, spoofedDiff) => {
+    // If channel is set and user already has bosses tracked
+    if (trackingChannel && bosses.hasOwnProperty(player)) {
+        // Compute diff for each boss
+        let diff;
+        try {
+            if (spoofedDiff) {
+                diff = spoofedDiff;
+                Object.keys(diff).forEach((boss) => {
+                    killCounts[boss] += diff[boss];
+                });
+            } else {
+                diff = computeDiff(bosses[player], killCounts);
             }
```

Comment:
This will break because - unlike skills - the boss counts don't seem to initialize everything at a base value. For skills, everyone starts off at 1, so the pre-existing level is a number. For bosses, it's trying to compute a diff between `undefined` and some number, resulting `NaN`. You need to handle the potentially undefined initial value for the spoofed case here but also in `computeDiff` (this function traverses over the "before" map's keys, but the "after" map may contain new keys).